torch/csrc/cuda/nccl.h

```diff
@@ -6,11 +6,18 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 #include <c10/util/Optional.h>
 
-#include <nccl.h>
-
 #include <cstddef>
 #include <vector>
 
+enum ncclDataType_t : unsigned int;
+enum ncclResult_t : unsigned int;
+enum ncclRedOp_t : unsigned int;
+typedef struct ncclComm* ncclComm_t;
+struct ncclUniqueId;
+int32_t ncclSum;
+
+
```

Comment:
Why `ncclSum` global variable is declared in this header? Also wouldn't it lead to a linker conflict?
```suggestion
int32_t ncclSum;


```

Reply:
ncclSum is one of the values in ncclRedOp_t.  I suppose I could replace the default with a hard coded value, I assume ncclRedOp_t::ncclSum is 0.